File: tests/fixtures/gcs-py/basic/no_imports.py
class FakeBlob:
    def download_as_text(self): return ""
    def upload_from_string(self, s): pass
    def delete(self): pass


class FakeBucket:
    def blob(self, key): return FakeBlob()


class FakeClient:
    def bucket(self, name): return FakeBucket()


client = FakeClient()


def local_fetch():
    return client.bucket("nope").blob("nope").download_as_text()


def local_save():
    return client.bucket("nope").blob("nope").upload_from_string("x")

File: tests/fixtures/gcs-go/basic/no_imports.go
package main

import "context"

type fakeReader struct{}
type fakeObject struct{}
type fakeBucket struct{}
type fakeClient struct{}

func (fakeObject) NewReader(_ context.Context) (*fakeReader, error) { return nil, nil }
func (fakeObject) Delete(_ context.Context) error                   { return nil }
func (fakeBucket) Object(_ string) *fakeObject                      { return &fakeObject{} }
func (fakeBucket) Delete(_ context.Context) error                   { return nil }
func (fakeClient) Bucket(_ string) *fakeBucket                      { return &fakeBucket{} }

func localFetch(ctx context.Context) {
	c := fakeClient{}
	_, _ = c.Bucket("nope").Object("nope").NewReader(ctx)
}
